content/en/tutorials/anim-blending.md

```diff
@@ -0,0 +1,107 @@
+---
+title: Anim Blending
+template: tutorial-page.tmpl.html
+tags: animation
+thumb: https://s3-eu-west-1.amazonaws.com/images.playcanvas.com/projects/12/405874/A8B1FE-image-75.jpg
+---
+
+<iframe src="https://playcanv.as/p/HI8kniOx/" ></iframe>
+
+*Click on screen to focus, then press the 'p' key to blend to a punch animation*
+
+This tutorial illustrates the basics of animation blending.
+Objects in your scene may be animated; machines or characters are good examples of things that you might want to animate. Generally, when 3D content is created, individual animations are authored and these animations are typically referred to as cycles (because they loop). For example, a human character could have an idle cycle, a walk cycle, a run cycle and so on. As a PlayCanvas developer, you'll want a mechanism to play these animations back on your animated object. Additionally, you do not want these animations to 'pop' as one is switched for another. To remedy this, you should use animation blending which implements a smooth transition from one animation to another. This dramatically improves the visual fidelity of your animated object.
+Let's examine how this is achieved via PlayCanvas...
+
+## The Animstategraph Asset
+
+In order to add animations to a model, we must first create an animstategraph asset. These assets control multiple animations associated with a single entity, determine when each of those animations should play and how they should be blended together.
+
+When creating an animstategraph asset, you’ll be presented with it’s editor:
+
+![Initial Editor][1]
+
+By default the animstategraph asset creates a single animation state called ‘Initial State’ which has a default transition from the START state to it. This means when the animstategraph first begins playing, it’ll immediately begin playing the Initial State. This can be renamed to Idle which is the name of the initial animation used in this tutorial. We can then right click the canvas and select `Add new state` to create another animation state in the graph. This state can be renamed to Punch:
```

Comment:
```suggestion
By default, the animstategraph asset creates a single animation state called ‘Initial State’ which has a default transition from the START state to it. This means when the animstategraph first begins playing, it’ll immediately begin playing the Initial State. This can be renamed to Idle which is the name of the initial animation used in this tutorial. We can then right click the canvas and select `Add new state` to create another animation state in the graph. This state can be renamed to Punch:
```